### ISSUE TITLE
robust_ocp.py broken

### ISSUE THREAD

**Issue by jaeandersson** [casadi/casadi#1459]
From travis:
```
Ran for 0.0 seconds
In ../docs/examples/python, robust_ocp.py failed:
==============================
returncode:  1
Traceback (most recent call last):
  File "robust_ocp.py", line 65, in <module>
    h_margin = [ sqrt(quad_form(jacobian(i,x).T,P)) for i in h_nom ]
  File "/usr/lib/python2.7/dist-packages/casadi/casadi.py", line 18123, in quad_form
    return _casadi.quad_form(*args)
RuntimeError: The assertion "isVector()" on line 1375 of file "/home/travis/build/casadi/casadi/./casadi/core/sx/../matrix/matrix_impl.hpp" failed. 
Please notify the CasADi developers.
==============================
03. accessing_sx_algorithm.py
```

The error message looks legit to me. @jgillis - any idea of what's going on?

**Comment by jgillis** [comment]
I'll have a look right now

**Comment by jgillis** [comment]
ftfy

**Comment by jgillis** [comment]
Nice, the example gives a segfault at destruction time. Inspecting..

**Comment by jgillis** [comment]
In 92459bd, it gives no segfault, just "CasADi warning: "Reference counting failure. Possible cause: Circular dependency in user code.""

**Comment by jaeandersson** [comment]
Yeah, something fishy alright.

**Comment by jgillis** [comment]
Segfault fixed in 0340517eaf326206261f77ec50b8d5747a3aa047
